pkg/controlplane/apiserver/server.go

```diff
@@ -139,7 +138,7 @@ func (c completedConfig) New(name string, delegationTarget genericapiserver.Dele
 	}
 	if len(c.SystemNamespaces) > 0 {
 		s.GenericAPIServer.AddPostStartHookOrDie("start-system-namespaces-controller", func(hookContext genericapiserver.PostStartHookContext) error {
-			go systemnamespaces.NewController(c.SystemNamespaces, client, s.VersionedInformers.Core().V1().Namespaces()).Run(hookContext.StopCh)
+			go systemnamespaces.NewController(c.SystemNamespaces, client, s.VersionedInformers.Core().V1().Namespaces()).Run(hookContext.Done())
```

Comment:
Thinking ahead: for [contextual logging](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/3077-contextual-logging) what we really want to pass around is the `context.Context` itself, not it's `Done` channel.

This cannot be done (sic) without API changes, so it's more work and leads to a bigger PR.

It then will need to touch the same lines again. Would it make sense to extend the scope of this PR? I'm fine either way, I just wanted to point out that this is not necessarily the final state.

Reply:
But this means we need to change each `Run` function for this? 
I would make them separate PRs if we can. Smaller is always better :D

Reply:
For private code, yes, `Run` would need to be changed.

For public code it's worse: most likely we will need a `RunWithContext` alternative.